overlord/configstate/configcore/log_control.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.system.printk.console-loglevel"] = true
+}
+
+func validateLogControlSettings(tr config.ConfGetter) error {
+	consoleLoglevelStr, err := coreCfg(tr, "system.printk.console-loglevel")
+	if err != nil {
+		return err
+	}
+	if consoleLoglevelStr != "" {
+		if n, err := strconv.ParseUint(consoleLoglevelStr, 10, 8); err != nil || (n < 0 || n > 7) {
+			return fmt.Errorf("loglevel must be a number between 0 and 7, not %q", consoleLoglevelStr)
+		}
+	}
+	return nil
+}
+
+func handleLogControlConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error {
+	root := dirs.GlobalRootDir
+	if opts != nil {
+		root = opts.RootDir
+	}
+	cfgPath := filepath.Join(root, "/etc/sysctl.d/10-console-messages.conf")
```

Comment:
We should use snapd in the filename to make it clear it's managed by us, e.g. "10-snapd-kernel-loglevel.conf", depending on the final name per my first comment.

Reply:
10-console-messages.conf already exists. We update it when we set new value by snap set.

Reply:
Ah, I see. I'm not sure what the best practice is, should we maybe create a file with higher priority then?

Reply:
I used 20-snap-console-messages.conf in the draft version. But, I was not sure if it's a good idea to have two config files for log level. So, I removed it in the final version.

Reply:
that file comes from and belongs to the procps package, we shouldn't touch it.  snapd should probably write a single file for all it's  sysctl needs:

`/etc/sysctl.d/99-snapd.conf